src/test/java/seedu/commands/EditCommandTest.java

```diff
@@ -0,0 +1,117 @@
+package seedu.commands;
+
+import seedu.classes.Constants;
+import seedu.classes.Parser;
+import seedu.type.IncomeList;
+import seedu.type.SpendingList;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class EditCommandTest {
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+    private final PrintStream originalErr = System.err;
+
+    private final IncomeList incomes = new IncomeList();
+    private final SpendingList spendings = new SpendingList();
+
+    @BeforeEach
+    public void setUp() {
+        Command c = Parser.parse("add spending 10 girlfriends");
+        c.execute(incomes, spendings);
+
+        c = Parser.parse("add spending 10 macdonalds");
+        c.execute(incomes, spendings);
+
+        c = Parser.parse("add income 10 savings");
+        c.execute(incomes, spendings);
+
+        c = Parser.parse("add income 10 dividends");
+        c.execute(incomes, spendings);
+
+        System.setOut(new PrintStream(outContent));
+        System.setErr(new PrintStream(errContent));
+    }
+
+    @AfterEach
+    public void restore() {
+        System.setOut(originalOut);
+        System.setErr(originalErr);
+    }
+
+    @Test
+    public void execute_noArgs_expectIllegalArgumentException() {
```

Comment:
Maybe its good to have a better test name, instead of `noArgs`, maybe like `missingEditDescription`